query_store.php

```diff
@@ -97,10 +97,19 @@
 if( isset( $t_filter['_source_query_id'] ) ) {
 	unset( $t_filter['_source_query_id'] );
 }
-$t_filter_string = filter_serialize( $t_filter );
 
-$t_new_row_id = filter_db_set_for_current_user( $t_project_id, $f_is_public,
-												$f_query_name, $t_filter_string );
+# Check that the user has permission to create stored filters
+if( !access_has_project_level( config_get( 'stored_query_create_threshold' ) ) ) {
+	access_denied();
+}
+
+# ensure that we're not making this filter public if we're not allowed
+if( !access_has_project_level( config_get( 'stored_query_create_shared_threshold' ) ) ) {
+	$f_is_public = false;
```

Comment:
Should this trigger access denied rather than changing the script parameters?  This case shouldn't typically happen anyways since form should do such validation.

Reply:
That code moved "as is" from `filter_db_set_for_current_user()`
you may be right

Reply:
ok, updated